src/jitlayers.cpp

```diff
@@ -189,7 +189,7 @@ static jl_callptr_t _jl_compile_codeinst(
             }
             jl_atomic_store_release(&this_code->invoke, addr);
         }
-        else if (this_code->invoke == jl_fptr_const_return_addr && !decls.specFunctionObject.empty()) {
+        else if (jl_atomic_load_relaxed(&this_code->invoke) == jl_fptr_const_return_addr && !decls.specFunctionObject.empty()) {
```

Comment:
Do we want to merge these two loads of the invoke pointer (this + line 183) into a variable?

Reply:
The token `jl_fptr_const_return_addr` is only set during construction, so it is constant. Indeed, the intent is for this value to be only set once, so repeated reads will give the same answer or NULL.